packages/react-native/React/Fabric/Mounting/ComponentViews/Image/RCTImageComponentView.mm

```diff
@@ -86,6 +121,23 @@ - (void)updateState:(const State::Shared &)state oldState:(const State::Shared &
 
   if (!havePreviousData ||
       (newImageState && newImageState->getData().getImageSource() != oldImageState->getData().getImageSource())) {
+    const auto &imageProps = static_cast<const ImageProps &>(*_props);
+    if (!imageProps.defaultSources.empty()) {
+      UIImage *defaultImage = [self _getUIImageWithImageSource:imageProps.defaultSources.front()];
+
+      if (defaultImage) {
+        __weak RCTImageComponentView *weakSelf = self;
+        [self _setImage:defaultImage
+            setImageBlock:^(UIImage *finalImage) {
+              RCTImageComponentView *strongSelf = weakSelf;
+              if (strongSelf) {
+                if (!strongSelf->_imageView.image) {
+                  strongSelf->_imageView.image = finalImage;
```

Comment:
Shouldn't we emit an `OnLoadEnd` event here?

Reply:
The same as old arch, we only fire image load events for image source.